File: solutions/HJK9810/2026-08-25/medium/5432.py
def algorithm():
    line = list(input())
    stack = []

    cut_count = 0
    is_cut_avail = False

    for alpa in line:
        if alpa == '(':
            is_cut_avail = True
            stack.append(alpa)
            continue

        if is_cut_avail:
            is_cut_avail = False
            stack.pop()
            cut_count += len(stack)
        else:
            stack.pop()
            cut_count += 1

    return cut_count

T = int(input())
# 여러개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for test_case in range(1, T + 1):
    print(f"#{test_case} {algorithm()}")

File: solutions/HJK9810/2026-08-25/medium/5432.java
import java.util.ArrayDeque;
import java.util.Scanner;

class Solution {
  private int algorithm (String inputStr, ArrayDeque<Character> stack) {
    char[] line = inputStr.toCharArray();
    stack = new ArrayDeque<>();
    int cutCount = 0;
    boolean isCutAvail = false;

    for (char alpa : line) {
      if (alpa == '(') {
        isCutAvail = true;
        stack.push(alpa);
        continue;
      }

      if (isCutAvail) {
        isCutAvail = false;
        stack.pop();
        cutCount += stack.size();
      } else {
        stack.pop();
        cutCount++;
      }
    }

    return cutCount;
  }

	public static void main(String args[]) throws Exception {
    Solution sol = new Solution();
		Scanner sc = new Scanner(System.in);
		int T = sc.nextInt();
    
    ArrayDeque<Character> stack = new ArrayDeque<>();

		for(int test_case = 1; test_case <= T; test_case++) {
      String line = sc.next();
      int cutCount = sol.algorithm(line, stack);

      System.out.printf("#%d %d\n", test_case, cutCount);
		}
	}
}
